### ISSUE TITLE
NPE in onPreDraw

### ISSUE THREAD

**Issue by TealOcean** [andkulikov/Transitions-Everywhere#50]
I've been getting reports of this exception from time to time showing up in our logs.

```
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.Object android.support.v4.util.ArrayMap.get(java.lang.Object)' on a null object reference
	 at com.transitionseverywhere.TransitionManager$MultiListener.onPreDraw(SourceFile:293)
	 at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:944)
	 at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2420)
	 at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1324)
	 at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6712)
	 at android.view.Choreographer$CallbackRecord.run(Choreographer.java:894)
	 at android.view.Choreographer.doCallbacks(Choreographer.java:696)
	 at android.view.Choreographer.doFrame(Choreographer.java:631)
	 at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:880)
	 at android.os.Handler.handleCallback(Handler.java:815)
	 at android.os.Handler.dispatchMessage(Handler.java:104)
	 at android.os.Looper.loop(Looper.java:207)
	 at android.app.ActivityThread.main(ActivityThread.java:5728)
	 at java.lang.reflect.Method.invoke(Native Method)
	 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:789)
	 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:679)
```

Looking at the code, I noticed this:

```
            // Add to running list, handle end to remove it
            final ArrayMap<ViewGroup, ArrayList<Transition>> runningTransitions =
                    getRunningTransitions();
            ArrayList<Transition> currentTransitions = runningTransitions.get(mSceneRoot);
```

It appears that `getRunningTransitions()` may be giving me a null pointer. When I looked at the code for this, I noticed it's using WeakReferences. It appears to be possible that the garbage collector is deleting the reference before the final call to `runningTransitions.get()` is completed, leading to the NPE I'm experiencing.

I haven't been able to find exact reproduction steps yet, but I find them I'll post them here.

**Comment by andkulikov** [comment]
Thanks!
Fixed in 1.7.0. AOSP bug created https://code.google.com/p/android/issues/detail?id=227305

**Comment by TealOcean** [comment]
You rock! 👍
